packages/automerge-repo-network-websocket/src/BrowserWebSocketClientAdapter.ts

```diff
@@ -55,24 +56,36 @@ export class BrowserWebSocketClientAdapter extends WebSocketNetworkAdapter {
     )
   }
 
+  private transmit(message: FromClientMessage) {
+    if (!this.socket || this.socket.readyState !== WebSocket.OPEN) {
+      throw new Error("Websocket Socket not ready!")
+    }
+
+    const encoded = CBOR.encode(message)
+    // This incantation deals with websocket sending the whole
+    // underlying buffer even if we just have a uint8array view on it
+    const arrayBuf = encoded.buffer.slice(
+      encoded.byteOffset,
+      encoded.byteOffset + encoded.byteLength
+    )
+
+    this.socket?.send(arrayBuf)
+  }
+
   join() {
     if (!this.socket) {
       throw new Error("WTF, get a socket")
     }
     if (this.socket.readyState === WebSocket.OPEN) {
-      this.socket.send(
-        CBOR.encode(joinMessage(this.peerId))
-      )
+      this.transmit(joinMessage(this.peerId!))
     } else {
       this.socket.addEventListener(
         "open",
         () => {
           if (!this.socket) {
             throw new Error("WTF, get a socket")
           }
-          this.socket.send(
-            CBOR.encode(joinMessage(this.peerId))
-          )
+          this.transmit(joinMessage(this.peerId!))
```

Comment:
i know i wrote this, but it would be nice if there were a better idiom for this "do it now or when you can" kind of thing. obviously there's await but that flavors this function with async